apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -286,3 +292,23 @@ function wpcom_display_global_styles_banner_extra_tooltip() {
 	<?php
 }
 add_action( 'launch_bar_extra_tooltip_toggle_global_styles', 'wpcom_display_global_styles_banner_extra_tooltip' );
+
+/**
+ * Invalidates the cached Global Styles after changing them or when previewing them.
+ */
+function wpcom_invalidate_global_styles_cache() {
+	$transient_name = 'gutenberg_global_styles_' . get_stylesheet();
+
+	if ( wpcom_should_limit_global_styles() && wpcom_is_previewing_global_styles() ) {
+		add_filter( 'pre_transient_' . $transient_name, '__return_null' );
+		add_filter( 'pre_set_transient_' . $transient_name, '__return_false' );
+	}
+
+	add_action(
+		'save_post_wp_global_styles',
+		function () use ( $transient_name ) {
+			delete_transient( $transient_name );
+		}
+	);
```

Comment:
Nitpick, this could be replaced by a slightly simpler:
```suggestion
	add_action( 'save_post_wp_global_styles', fn() => delete_transient( $transient_name ) );
```